### ISSUE TITLE
MailChimp connected sites List Settings syncing seemingly forever

### ISSUE THREAD

**Issue by jordanbrauer** [mailchimp/mc-magento2#66]
## Preconditions:

1. Magento CE `2.1.5`
2. MailChimp for Magento 2 `1.0.22`
3. PHP `7.0.X`

## Steps to reproduce:

1. Install MailChimp for Magento 2 with `composer require mailchimp/mc-magento2 1.0.22`.
2. Configure MailChimp for Magento 2 following [this guide](http://kb.mailchimp.com/integrations/e-commerce/connect-or-disconnect-mailchimp-for-magento-2).
3. Ensure that the "Enable Two Way Sync" option is enabled.
4. Wait for the Magento customers to sync to the assigned MailChimp list.
5. Wait for the Magento products to sync to the assigned MailChimp list.

## Expected result:

* The Carts sync properly for automated emails such as the abandoned cart email to be used.
* The List Settings section in the Connected Sites MailChimp page finishes syncing.

## Actual result:

* Carts don't sync
* The List Settings section in the Connected Sites MailChimp page syncs seemingly forever.

**MailChimp > Connected Sites > List Settings**
![image](https://user-images.githubusercontent.com/18744334/29223345-e113fd4c-7e8b-11e7-8ca1-d7c00320a935.png)

## Additional information:

Here are some screenshots from Magento and MailChimp to show that our products and cutomers have initially sync'd (we still have to test creating a new product and customer syncing). 

**Magento > Stores > Configuration > MailChimp > MailChimp > MailChimp General Configuration**
![image](https://user-images.githubusercontent.com/18744334/29223472-46df6c38-7e8c-11e7-9047-968bbba848d4.png)

**MailChimp > Templates > Content Manager > Products**
![image](https://user-images.githubusercontent.com/18744334/29223521-7af6b102-7e8c-11e7-939c-9aeae81cdd10.png)

**Magento > Products > Catalog**
![image](https://user-images.githubusercontent.com/18744334/29223581-b2d14984-7e8c-11e7-9138-bcb249761dc1.png)

**Magento > Customers > All Customers**
![image](https://user-images.githubusercontent.com/18744334/29223596-c3ce86ac-7e8c-11e7-8a88-e5870ae55140.png)

If this has all synced, why is the MailChimp list continuing to sync and never end?

**Comment by gonzaloebiz** [comment]
Hi @jordanbrauer 

we are working on this with mailchimp

**Comment by jordanbrauer** [comment]
Sounds good.

Thanks @gonzaloebiz.

**Comment by takmonster** [comment]
Hey there,

I'm a dev at MailChimp. We've dug into this a bit and have a fix in mind that we hope to have out in the coming weeks. I'll update here once it's live.

---

On an implementation-side, this seems to happen when you create a Store via the API with `is_syncing` set to `false` and then never manipulate that value again. As long as you don't manipulate that value, the app considers the store to be "syncing" (even though it shouldn't).

This is a bit odd because I'd expect the integration to always be setting `is_syncing` to `true` at some point in the connection process - so I'm not quite sure how this is occurring within the integration's flow. However, this is still a display issue we've narrowed down on our end and should have fixed soon.

**Comment by nickalmeida** [comment]
Hey @takzilla any updates on this issue or any workaround? I am desperate to get the abandoned emails to work with no luck. Please help

**Comment by takmonster** [comment]
The fix isn't live yet I'm afraid. In the mean time, would you be able to send me an email including the email address that's associated with your MailChimp account? I'd like to confirm that our fix will resolve the issue for you once it is live. If anyone else besides @nickalmeida is seeing this, feel free to send me an email with the same info as well.

**Comment by sam110386** [comment]
![image](https://user-images.githubusercontent.com/16823364/31277914-c1a0c178-aac0-11e7-82a0-e11636543530.png)
We are also facing same issue. We installed MailChimp v1.0.23. Please help if you resolved this issue.

**Comment by epiceye** [comment]
We are experiencing the same issue - I found the following error in the cron_schedule table - not sure if it is related. "error   | Requested country is not available. "
I am running magento V2.1.0 and PHP 7.0.x and mc-magento 1.0.24
![01](https://user-images.githubusercontent.com/8456151/31542040-cebf2372-b010-11e7-8bd8-d7cdadf5e506.png)

**Comment by ghost** [comment]
We seem to face the same issue. Everything is syncing (Customers, Products, Orders and carts) but when it was finished Mailchimp still said that the list is syncing.

**Comment by jordanbrauer** [comment]
@takzilla Any update on the issue?

**Comment by ghost** [comment]
is there a workaround so we can set the sync to complete manually?
Really need this working as we can't sync any store at the moment

**Comment by takmonster** [comment]
@jordanbrauer @marcmayo 

Can you both send me an email with the email address associated with your MailChimp accounts? I can take a closer look. My email contact should be visible from my profile page.

**Comment by lgrassini** [comment]
Hello @takzilla @gonzaloebiz 

We are facing the same problem here. We can't run our abandoned cart strategies since is not working.

@takzilla I'll send you my mailchimp account information in a pm.

**Comment by gonzaloebiz** [comment]
Hi @lgrassini 

this issue will be fixed next version when [this](https://github.com/mailchimp/mc-magento2/issues/97) applied.

Best

**Comment by moneef** [comment]
Hey @gonzaloebiz thank you for the great work, 
i'm having the same issue , can you tell us when will the fix will be implemented ?

**Comment by NiteshKuyate** [comment]
We seem to face the same issue. Everything is syncing (Customers, Products, Orders and carts) in Magento admin.
Also MailChimp > Connected Sites > List Settings > Syncing list:#### continuesly.
@gonzaloebiz When next version will launch aprox.?

**Comment by KevinMace** [comment]
We're also having a very similar issue. The sync has been going for days, with only our client's product catalog actually synced and none of their customer/order/cart data.

<img width="418" alt="screen shot 2017-10-27 at 10 07 02" src="https://user-images.githubusercontent.com/1065240/32096586-cab8fd54-bafe-11e7-84f5-e8577b0f8010.png">

**Comment by moneef** [comment]
@KevinMace and @NiteshKuyate ,

you can use `composer require "mailchimp/mc-magento2" "dev-develop"` to get the latest version with the fix, also you should change composer `"minimum-stability" ` to `"dev"` before using the command

**P.S i don't recommend getting the dev version**

##  **also guys if you have time i need help with mailchimp and how to manage my lists for two store-views, i read the knowledge base but i didn't get it !**
![1](https://user-images.githubusercontent.com/6404057/32097320-d59932c6-bb11-11e7-9afb-1ce83545668f.PNG)
![2](https://user-images.githubusercontent.com/6404057/32097396-18bbe666-bb12-11e7-9e01-b61637eb4c4a.PNG)

**Comment by gonzaloebiz** [comment]
@moneef @NiteshKuyate next week i think

**Comment by NiteshKuyate** [comment]
@gonzaloebiz We are waiting to make site live. Can you tell aprox or exact launching date for updated version ?

**Comment by ghost** [comment]
@gonzaloebiz thanks for fixing this. @moneef You mentioned earlier that it's possible to install a developer version with the fix? Can we also apply this fix by replacing files instead of using the composer? 
We are really running out of time as we want to start early bird Christmas campaigns soon etc.

**Comment by moneef** [comment]
@marcmayo check this out https://github.com/mailchimp/mc-magento2/commit/7f5b0b96b127783c5079dd84cef2b855e806d6ba , https://github.com/mailchimp/mc-magento2/commit/a94bf69f77bae14fd5b8a5ca89012e4c88d8e938 and https://github.com/mailchimp/mc-magento2/commit/91b7db721f204943c54e2751e844abb8604fc708

It's better to get the full version 1.0.25 from composer as I mentioned before, it has alot of improvement on 1.0.24, it's much faster to update using composer than editing files.

**Comment by lgrassini** [comment]
Hi guys, 
Any news about when this going to be released to the stable branch?
Thanks.

@gonzaloebiz @takzilla

**Comment by NiteshKuyate** [comment]
Hello ,
We have installed develop version **1.0.25** by downloading .zip from **github** then configured it to test mailchimp account having only 10 subscribers before 15 hours but still same issue exists i.e. **List Settings syncing seemingly forever.**
@takzilla @gonzaloebiz @moneef

**Comment by gonzaloebiz** [comment]
Hi @NiteshKuyate 

you already look at the logs?

**Comment by NiteshKuyate** [comment]
@gonzaloebiz 
you already look at the logs?
P.S. > Yes, nothing found in log.
Can you please tell, why this bug closed and added "Done" label ?

**Comment by ghost** [comment]
@gonzaloebiz @takzilla @moneef 
unfortunately we are still having the same issue. Store is syncing forever.
We are using 1.0.25 and have setup a store with only around 300 customers.
What we noticed is that the sync says 

Total Customers: 384 -> There are only 306 customers in that store checking on Customers-> All Customers
Total Products: 4 -> is correct
Total Orders: 77 -> is correct
Total Carts: 5 -> can't tell if it's correct

We really really need this working as Christmas is our main season and our marketing suffers from that.
I could provide access to my staging environment and we could sort this out maybe?

**Comment by gonzaloebiz** [comment]
Hi @marcmayo 

take a look at the cron_schedule table for job_code ebizmarts_ecommerce

**Comment by archonkulis** [comment]
@gonzaloebiz 

There were some problems within that table - all entries were with status "running" - looks like it encountered some errors or something like that.

Removed all entries with code `ebizmarts_ecommerce` and on the next run received "`Resource not found`" https://prnt.sc/ha7fab . Although, the data seem to be synced anyway. Probably because it first takes correct store ids.

In the file `app/code/Ebizmarts/MailChimp/Cron/Ecommerce.php`

If I change 148th line from:

`            if($val&&!$this->_helper->getConfigValue(\Ebizmarts\MailChimp\Helper\Data::XML_PATH_IS_SYNC."/$mailchimpStoreId", 0, 'default')) {
`

to:
`
            if($val&&!$this->_helper->getConfigValue(\Ebizmarts\MailChimp\Helper\Data::XML_PATH_IS_SYNC."/$mailchimpStoreId", 0, 'default')&&$mailchimpStoreId != -1) {`

Then all is good - the status is success and there is no error.

We have multi-store Magento. Only one store for testing purposes is connected to MailChimp. All other stores have "Enabled" set to "No" in the Mailchimp General Configuration.

**Comment by jsdupuis** [comment]
Any update on this issue? Facing the same problem on Magento 2.1.8 with MailChimp 1.0.25.

**Comment by SantiParagarino** [comment]
Hello @jsdupuis 

We have seen all the cases fixed since version 1.0.25. I think your case is a different situation therefore I suggest you to contact our support at mailchimp@ebizmarts-desk.zendesk.com

Regards.

**Comment by alloylab** [comment]
I am having this issue as well... with no data ever displaying in MailChimp...

Magento 2.2.1
mc-magento2 1.0.26

**Comment by ghost** [comment]
We are also having the same issues as well. Mailchimp List Settings does not get past 'syncing lists'  It has been spinning for 3 days.  No product data is displaying in mailchimp and magento's mailchimp general panel shows: 

Total Subscribers: 4818
Ecommerce Data uploaded to MailChimp
Total Customers: 0
Total Products: 0
Total Orders: 0
Total Carts: 0
This account is currently syncing

Magento 2.2.1
MailChimp v1.0.25

**Comment by ACassiusD** [comment]
Does anyone have a solution for this yet. Same issue. Stuck on syncing. Contacted Mailchimp support and they told me that no syncing data is being passed and asked me to contact ebixzmarts support.

**Comment by gonzaloebiz** [comment]
@ACassiusD 

take a look at the cron_schedule table, if the jobs are with status running check your cli php configuration, generally is a insufficient memory issue (not a extension issue)

Best

**Comment by cefrigo** [comment]
Facing same issue here, Sync never ends. Both crons are now running fine after heavy debugging but sync seems to be stuck any safe way to restart it ?

**Comment by TomsBugna** [comment]
I am facing the same issue. Magento version: CE 2.2.3. Extension version: 1.0.28. Any updates on how to fix this?

**Comment by lgrassini** [comment]
Hey @Tomijs, @cefrigo and others experiencing this issue...

I contacted the Mailchimp support regarding this several weeks ago. I was able to fix the problem following this directions the Mailchimp team provided me: 

"_Hey there,_ 

_My name is Michael and I'm an E-Commerce specialist here at MailChimp. I'm in the process of reviewing your store's connection and I noticed that MageMonkey for Magento 2 is still connected on the backend in your MailChimp account._ 

_Can you confirm with your Magento developers that MageMonkey has been completely removed from your Magento 2 store? MageMonkey has been known to cause numerous conflicts with MailChimp for Magento 2 and we will need to ensure that there are no traces of MageMonkey remaining in your store._ 

_Once you confirm that it's been removed, then I would recommend removing the API connection from within your MailChimp account by going to account > extras > API Keys > scroll down to the authorized applications area > select the X to the right of any MageMonkey connection on that page._ "

**TL;DR**: 1) Check that you have completely removed the MageMonkey extension. 2) Remove any API keys being granted to MageMonkey. 3) Re-connect your mc-magento2 extension with Mailchimp.

**Comment by GideonBabu** [comment]
@gonzaloebiz @takzilla 

I am using this extension v1.0.28 and Magento version is 2.2.1

I am still facing this issue. Syncing the list is never ending process. Do you guys have a workaround for this? 
![screenshot-us18 admin mailchimp com-2018 05 03-16-09-07](https://user-images.githubusercontent.com/20984313/39572390-9bc74fb4-4eec-11e8-82e1-a57d601841d1.jpeg)

I am working on localhost and my cron_schedule table is empty

**Comment by gonzaloebiz** [comment]
Open a console and run

php --info | grep memory_limit

and be sure that the memory is enough to run magento

Best

**Comment by TomsBugna** [comment]
I can approve that @lgrassini solution works. Just with some adjustments, for example, make sure that before installing again the extension, database tables and data are removed as well. Make sure all files are removed, etc.

Store sync finished but now there is another issue - I can see that not all of the orders are synced to the MailChimp. Some of the completed orders and its products are not added to the customers data. But most of the data is synced correctly. Any reason for that?

Logs are without any errors in both places - DB and var/log directory.

**Comment by GideonBabu** [comment]
@gonzaloebiz I have run the command and attached the screenshot for your reference. 

![php --help](https://user-images.githubusercontent.com/20984313/39578730-8e619620-4f02-11e8-8dca-8509154c420e.png)

Does this extension works in localhost?

@gonzaloebiz Should we set up any cron job in the server to make the sync happen? or how does it work?

**Comment by gonzaloebiz** [comment]
@GideonBabu of course, you need to [setup the magento cron system](http://devdocs.magento.com/guides/v2.2/config-guide/cli/config-cli-subcommands-cron.html)

and yes, the extension works in localhost, not fully functional (the webhooks not work), but definitively works

Best

**Comment by caww** [comment]
Hi
Php 7.1
Magento 2.2.4
mailchimp 1.0.28
Facing Same issue in our shop
https://winesellersdirect.com.au/
please help us
name of mailchimp account is jasonborg

**Comment by EthicalErik** [comment]
Same problem here. It's stuck on "This account is currently syncing" and does nothing.
Magento 2.2.3 (preparing to upgrade to 2.2.4)
Ebizmarts Mailchimp 1.0.28

I tried the solution provided by lgrassini however this did nothing for us.

**Comment by gonzaloebiz** [comment]
If you have problems with this, is not an extension issue, is typically a configuration error in your server, so please contact our support team at mailchimp@ebizmarts-desk.zendesk.com

Best
